.github/DEVELOPMENT.md

```diff
@@ -239,3 +239,24 @@ re-compiles when changes to source files are detected:
 To iterate quickly, simply re-build the project in IntelliJ after packaging is
 complete. Project resources will be hot-reloaded and changes are reflected on
 browser refresh.
+
+## Releases
+
+Trino aims for frequent releases, generally once per week. This is a goal but
+not a guarantee, as critical bugs may lead to a release being pushed back or
+require an extra emergency release as a patch.
+
+At the start of each release cycle, a GitHub issue is filed and pinned to track
+all necessary release notes. For example, see [the issue for Trino 395](https://github.com/trinodb/trino/issues/13913).
+In addition, a release notes pull request is updated and maintained throughout
+the week, tracking all merged commits to ensure every change is properly
+documented and noted. This uses the [release note template](../docs/release-template.md),
+with changes in each section arranged to have new features first, performance
```

Comment:
Should this be split into a list?

Reply:
I don't think so, I can't think of a way to make it a reasonable list that isn't super short or jarring.